Release browser hail tables

rileyhgrant opened the pull request:
Resolves #1369 

Adds a help page (à la the [v4 hail table help page](https://gnomad.broadinstitute.org/help/v4-hts)) that describes the gnomAD browser hail tables (variants, gene models).

Reviewing this PR mostly entails taking a glance over the help text to see if there's anything incorrect or misleading.

rileyhgrant commented:
Files to sync are:
- `gs://gnomad-v4-data-pipeline/output/genes/genes_grch37_annotated_5.ht`
- `gs://gnomad-v4-data-pipeline/output/genes/genes_grch38_annotated_5_no_constraint.ht`
- `gs://gnomad-v4-data-pipeline/output/gnomad_v4/gnomad_v4_variants_annotated_4.ht`

nadeaujoshua commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+  - `mean`: Mean depth of coverage at this locus.
+  - `median`: Median depth of coverage at this locus.
+  - `over_1`: Percentage of samples with a coverage greater than 1 at this locus.
+  - `over_5`: TPercentage of samples with a coverage greater than 5 at this locus
```
Just flagging typo here.

nadeaujoshua commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+- `gene_version`: Gene version.
+- `gencode_symbol`: GENCODE gene symbol.
+- `chrom`: Chromosome in which gene is located.
+- `strand`:Gene strand.
```
I think another typo.

nadeaujoshua reviewed (approved):
Hey this looks good. I flagged what looks like a few typos.

mattsolo1 commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+title: 'gnomAD v4 Browser Hail Tables'
+---
+
+To go from the gnomAD release tables to a format that is more convenient for the browser to display, the input tables undergo a few transformations. This groups data in ways that are more convenient to query from the database these tables ultimately get loaded into. In addition to performing some reshaping of the data, additional annotations or information is added as needed.
```
I wonder if we can we add a bit more high level explanation about these tables? How about:

---

We release two additional data tables to the public so that our users can more easily incorporate gnomAD data into external pipelines and analyses in a manner consistent with what they see in the browser.

## gnomAD v4.1 exome/genome variant table

To convert the standard gnomAD variant release tables into a format more suitable for browser display, we join the exome and genome tables on locus/allele to create a single table. This process ensures that they share the same site-level annotations, thus saving space and optimizing database/API queries. Additionally, frequencies are structured in a JSON-like format that is more easily consumable by web applications.

Each row (i.e., variant) in this table will have distinct allele frequency information and quality metrics depending whether it was present in the exome or genome callsets, but will share common annotations such as [VEP annotations](https://useast.ensembl.org/info/docs/tools/vep/index.html) and _in silico_ predictors.

The script for how this table is created can be found [here](
https://github.com/broadinstitute/gnomad-browser/blob/main/data-pipeline/src/data_pipeline/pipelines/gnomad_v4_variants.py).

## gnomAD 4.1 genes table

The second table underlies the gene models data seen in the browser, which contains detailed information on exon-coding regions, transcripts, identifiers, gene constraint, and co-occurrence data. The data from this table is derived from [GENCODE 39](https://www.gencodegenes.org/human/release_39.html), the [HUGO Gene Nomenclature Committee (HGNC)](https://www.genenames.org/), [MANE transcripts](https://www.ncbi.nlm.nih.gov/refseq/MANE/), [GTEx](https://gtexportal.org/home/) (coming soon), and from gnomAD secondary analyses.

The script for how this table is created can be found [here](https://github.com/broadinstitute/gnomad-browser/blob/main/data-pipeline/src/data_pipeline/pipelines/genes.py).

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+  - `transcript_version`: Transcript version.
+  - `gene_version`: Gene version.
+  - `is_mane_select`: Whether transcript is the MANE select transcript.
+  - `is_mane_select_version`: MANE Select version, has a value if this transcript is the MANE select transcript.
```
```suggestion
  - `is_mane_select_version`: MANE Select version; field has a value if this transcript is the MANE select transcript.
```

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+- `chrom`: Chromosome in which gene is located.
+- `strand`:Gene strand.
+- `start`: Gene genomic start position (position only).
+- `stop`: Gene genomic stop position (position only.- `xstart`: Gene genomic start position (format: chromosomeposition).Note that chrX is encoded as 23, chrY as 24, and chrM as 25.
```
```suggestion
- `stop`: Gene genomic stop position (position only).
- `xstart`: Gene genomic start position (format: chromosomeposition). Note that chrX is encoded as 23, chrY as 24, and chrM as 25.
```
this is pretty self explanatory, but it might be nice to have an example here as well

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+  - `reference_genome`: Reference genome associated with this transcript.
+  - `refseq_id`: Transcript RefSeq identifier.
+  - `refseq_version`: RefSeq version.
+- `hgnc_id`: GeneHGNC identifier.
```
```suggestion
- `hgnc_id`: HGNC gene identifier.
```

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+  - `ensembl_version`: Ensembl version.
+  - `refseq_id`: Transcript RefSeq ID.
+  - `refseq_version`: RefSeq version.
+- `preferred_transcript_id`: Transcript shown on the gene page by default. Field containsMANE Select transcript ID if it exists, otherwise contains Ensemblcanonical transcript ID.
```
```suggestion
- `preferred_transcript_id`: Transcript shown on the gene page by default. Field contains MANE Select transcript ID if it exists, otherwise contains Ensembl canonical transcript ID.
```

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+  - `refseq_id`: Transcript RefSeq ID.
+  - `refseq_version`: RefSeq version.
+- `preferred_transcript_id`: Transcript shown on the gene page by default. Field containsMANE Select transcript ID if it exists, otherwise contains Ensemblcanonical transcript ID.
+- `preferred_transcript_source`: Source of transcript ID used for `preferred_transcript_id` field; either "`mane_select`" or "`ensemble_canonical`".
```
```suggestion
- `preferred_transcript_source`: Source of transcript ID used for `preferred_transcript_id` field; either "`mane_select`" or "`ensembl_canonical`".
```

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+  - `syn_z`: Z-score for rare synonymous variants.
+  - `pli`: Probability of being loss-of-function intolerant (pLI) score.
+  - `flags`: Set containing constraint flags for transcript.
+- `heterozygous_variant_cooccurrence_counts`: Array containing information about heterozygous variant co-occurrence counts.
```
```suggestion
- `heterozygous_variant_cooccurrence_counts`: Array containing information about heterozygous variant co-occurrence counts. Struct is only present on the GRCh37 Hail Table.
```

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+  - `in_trans`: Count of variants in trans.
+  - `unphased`: Count of unphased variants.
+  - `two_het_total`: Total count of two heterozygous variants.
+- `homozygous_variant_cooccurrence_counts`: Array containing information about homozygous variant co-occurrence counts.
```
```suggestion
- `homozygous_variant_cooccurrence_counts`: Array containing information about homozygous variant co-occurrence counts. Struct is only present on the GRCh37 Hail Table.
```

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+
+#### gnomAD v4.1. browser gene models Hail Table annotations
+
+Row fields:
```
just checking that the globals are removed after full pipeline has been run -- I see this in `gs://gnomad-v4-data-pipeline/output/genes/genes_grch38_annotated_5.ht`:
```
----------------------------------------
Global fields:
    'annotations': struct {
        'canonical_transcript': struct {

        }
        'mane_select_transcript': struct {
            'version': str
        }
    }
----------------------------------------
```

ch-kr replied:
sorry, I worded that comment poorly -- it might be helpful to keep this global
```
- `mane_select_version`: MANE Select version used to annotate variants.
```
on this table as well (since this specifically contains gene information)

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+  - `flags`: A set containing the flags about the region this variant falls in.
+  - `freq`: A struct containing variant frequency information for each subset.
+  - `all`: Struct containing variant frequency information calculated across all samples.
+    - `ac`: The alternate allele count for this variant calculated across high-quality genotypes (genotypes with depth >= 10, genotype quality >= 20 and minor allele balance > 0.2 for heterozygous genotypes).
```
```suggestion
    - `ac`: The alternate allele count for this variant calculated across high-quality genotypes (genotypes with depth >= 10, genotype quality >= 20 and minor allele balance > 0.2 for heterozygous genotypes). This is the allele count displayed in the gnomAD browser (not `ac_raw` below).
```

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+      - `homozygote_count`: Number of homozygous alternate individuals in this genetic ancestry group.
+    - `non_ukb`: Struct containing variant frequency information from the non-UKB subset. Includes same fields as above struct (`all`).
+  - `faf95`: Struct containing information about the filtered allele frequency (FAF; 95% confidence interval [CI]).
+    - `grpmax`: Genetic ancestry group maximum (groupmax) FAF.
```
```suggestion
    - `grpmax`: Genetic ancestry group maximum (grpmax) FAF.
```

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+    - `non_ukb`: Struct containing variant frequency information from the non-UKB subset. Includes same fields as above struct (`all`).
+  - `faf95`: Struct containing information about the filtered allele frequency (FAF; 95% confidence interval [CI]).
+    - `grpmax`: Genetic ancestry group maximum (groupmax) FAF.
+    - `grpmax_gen_anc`: Genetic ancestry group associated with groupmax FAF.
```
```suggestion
    - `grpmax_gen_anc`: Genetic ancestry group associated with grpmax FAF.
```

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+    - `gnomad`: Struct containing information about the fafmax for all of gnomad for the joint exome and genome data.
+      - `faf95_max`: Max FAF value for the (95% CI).
+      - `faf95_max_gen_anc`: Genetic ancestry group associated with the grpmax FAF (95% CI).
+      - `faf99_max`: Max FAF(99% CI).f
```
```suggestion
      - `faf99_max`: Max FAF (99% CI).
```

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+  - `all`: An array containing colocated variants that are present in the entire exome dataset.
+  - `non_ukb`: An array containing colocated variants that are present in the non-UK Biobank (UKB) subset of the dataset.
+  - `subsets`: A set containing the subsets this variant is seen in.
+  - `flags`: A set containing the flags about the region this variant falls in.
```
```suggestion
  - `flags`: A set containing the flags about the region this variant falls in. See `region_flags` description on v4 Hail Tables [help page](https://gnomad.broadinstitute.org/help/v4-hts).
```

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+      - `n_larger`: Number of individuals with a higher age than the highest bin.
+    - `hom`: Struct containing age distribution information for individuals homozygous for this variant. Structured to allow easy histogram creation. Contains same fields as `het` above.
+  - `filters`: Set containing variant QC filters.
+  - `quality_metrics`: Struct containing variant quality metric histograms information.
+    - `allele_balance`: Struct containing variant allele balance histograms information.
+      - `alt_adj`: Struct containing variant allele balance information calculated across high-quality genotypes. Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`). This data is displayed in the "Allele balance for heterozygotes" histogram in the browser's variant page.
+      - `alt_raw`: Struct containing variant allele balance information calculated across unadjusted genotypes. Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`).
+    - `genotype_depth`: Struct containing information used to display genotype depth (DP) histograms.
+      - `all_adj`: Struct containing DP information calculated using high-quality genotypes. Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`).
+      - `all_raw`: Struct containing DP information calculated across unadjusted genotypes. Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`).
+      - `alt_adj`: Struct containing DP information calculated using high-quality genotypes (variant carriers only). Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`).
+      - `alt_raw`: Struct containing DP information calculated across unadjusted genotypes (variant carriers only). Structured to allow ease of creating histograms. Contains same fields as other histogram structs (e.g. `het`)
+    - `genotype_quality`: Struct containing information used to display genotype quality (GQ) histograms.
+      - `all_adj`: Struct containing GQ information calculated using high-quality genotypes. Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`).
+      - `all_raw`: Struct containing GQ information calculated across unadjusted genotypes. Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`).
+      - `alt_adj`: Struct containing GQ information calculated using high-quality genotypes (variant carriers only). Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`).
+      - `alt_raw`: Struct containing GQ information calculated across unadjusted genotypes (variant carriers only). Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`).
```
these descriptions are informative but a bit wordy (apologies if I suggested this previously). how about replacing
> Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`).

in each with just
>  Contains same fields as other histogram structs.

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+    - `value`: Metric value.
+- `genome`: Struct containing information about this variant from genome data. Contains all the same fields as the exome data, with the exception that the subsets are (`all` `hgdp`, `tgp`) instead of (`all`, `non_ukb`).
+- `rsids`: dbSNP reference SNP identification (rsID) numbers.
+- `in_silico_predictors`: Variant prediction annotations. Struct contains prediction scores from multiple in silico predictors for variants that are predicted to be missense, impacting protein function, evolutionarily conserved, or splice-altering. We chose scores for either MANE Select or canonical transcripts if a prediction score was available for multiple transcripts.
```
perhaps you could point users to this section (`in_silico_predictors`) on the v4 HTs help page here? would it be helpful to create hyperlinks to sections of the v4 HTs help page that contain fields that are also described here (e.g., `region_flags`, `in_silico_predictors`, `filters`)?

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+  - `grpmax_gen_anc`: Genetic ancestry group associated with the value `grpmax` above.
+- `faf99_joint`: A struct containing joint (exome + genome) FAF (99% CI). - `colocated_variants`: Array containing variants that are located at the same locus as this variant.
+- `coverage`: Struct containing coverage information for locus.
+  - `exome`: Struct containing exome coverage information.
```
should the fields beneath this (from `mean` to `over_100`) be indented?

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+  - `over_50`: Percentage of samples with a coverage greater than 50 at this locus.
+  - `over_100`: Percentage of samples with a coverage greater than 100 at this locus.
+  - `genome`: Struct containing genome coverage information. Contains the same fields as `exome` above.
+  - `transcript_consequences`: Array containing variant transcript consequence information.
```
should the fields beneath this be indented as well?

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+Row fields:
+
+- `interval`: Struct representing start and end positions of gene.
+- `gene_id`: Unique gene ID.
```
this is the ensembl gene ID, right? (might be helpful to include)

ch-kr reviewed (changes requested):
thank you for adding! I've added some minor comments and suggestions, mostly to help clarify the descriptions

rileyhgrant commented on data-pipeline/src/data_pipeline/data_types/coverage.py:
```diff
@@ -13,7 +13,7 @@ def prepare_coverage(coverage_path: str, filter_intervals: Optional[List[str]] =
     # Specific to v4, the data is contained in a coverage_stats field and we take the first element
     if "coverage_stats" in coverage_fields:
-        if type(coverage.coverage_stats) == hl.expr.expressions.typed_expressions.ArrayStructExpression:
+        if type(coverage.coverage_stats) is hl.expr.expressions.typed_expressions.ArrayStructExpression:
```
Honestly, this change was just because the linter we use, [`ruff`](https://docs.astral.sh/ruff/linter/) was complaining about these lines and it can be nice to just defer to a linter/formatter for stylistic decisions. 

With the change, they _should_ be equivalent seeing as its checking a type, so the underlying types of the two should be both the same address in memory (what `is`) checks for, and the same value (what `==` checks for)

rileyhgrant commented:
Thanks, all, for the comments. Going to be working through them right now.

Thanks especially to KC for your patience, and for going through this whole blurb of text multiple times now.

rileyhgrant commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,223 @@
+  - `phylop`: Base-wise conservation [score](https://pubmed.ncbi.nlm.nih.gov/37104612/) across the 241 placental mammals in the Zoonomia project. Score ranges from -20 to 9.28, and reflects acceleration (faster evolution than expected under neutral drift, assigned negative scores) as well as conservation (slower than expected evolution, assigned positive scores).
+  - `sift_max`: [Score](https://www.nature.com/articles/nprot.2009.86) reflecting the scaled probability of the amino acid substitution being tolerated, ranging from 0 to 1. Scores below 0.05 are predicted to impact protein function. We prioritize max scores for MANE Select transcripts where possible and otherwise report a score for the canonical transcript.
+  - `polyphen_max`: [Score](https://www.nature.com/articles/nmeth0410-248) that predicts the possible impact of an amino acid substitution on the structure and function of a human protein, ranging from 0.0 (tolerated) to 1.0 (deleterious). We prioritize max scores for MANE Select transcripts where possible and otherwise report a score for the canonical transcript.
+- `variant_id`: gnomAD variant ID.- `faf95_joint`: A struct containing joint (exome + genome) FAF information (95% CI).
```
Ah, this field `faf95_joint` is the one thats its own seperate struct outside of `exome` or `genome`. The `genome` struct just has some placeholder information saying "pretty much same exact shape as the exomes above".

I've made some indentation mistakes that have definitely made this more confusing, thanks for catching those.

ch-kr commented on data-pipeline/src/data_pipeline/data_types/gene.py:
```diff
@@ -226,6 +226,12 @@ def import_hgnc(path):
 
+def prepare_table_for_release(genes_path):
+    ds = hl.import_table(genes_path)
+    ds = ds.select_globals()
```
it might be helpful to retain the MANE select version on this table since it contains the gene model information

ch-kr reviewed (changes requested):
a few more minor things -- thanks for your patience with my nitpicks!

rileyhgrant commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,238 @@
+  - `faf99`: Struct containing information about the FAF (99% CI). Contains same subfields as above (`faf95`).
+  - `fafmax`: Struct containing information about the maximum FAF.
+    - `gnomad`: Struct containing information about the fafmax for all of gnomad for the exome data.
+      - `faf95_max`: Max FAF value for the (95% CI).
+      - `faf95_max_gen_anc`: Genetic ancestry group associated with the grpmax FAF (95% CI).
+      - `faf99_max`: Max FAF(99% CI).
+      - `faf99_max_gen_anc`: Genetic ancestry group associated with the max FAF (99% CI).
```
Yeah, this was a bit of a last minute transformation we did to allow our API code to look for the same fields on the v2 dataset as on the v4 dataset, so its currently duplicated.

I can make an issue that logs all the odd duplications like this to remove before the next big dataload, if that works? Alternatively we could temporarily put in a last step that cleans up this table for release, and have an associated issue to clean it up in th earlier steps.

ch-kr replied:
is it messy to do the second option (temporarily adding a step to clean up the table for release + creating a second issue to update the pipeline to clean up duplicate fields upstream)? removing duplicate fields will make this documentation clearer for the users, and we ideally won't store duplicate fields in a future release (e.g., in v5).

that said, I want to make sure this second option isn't too unbalanced in terms of effort -- is this second option also significantly more work?

rileyhgrant replied:
Sweet, thanks for the thoughts, we'll go with the second option. It'll make for a better table in the short and long term.

rileyhgrant commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,238 @@
+  - `grpmax`: Groupmax FAF value for all genetic ancestry groups across exomes + genomes.
+  - `grpmax_gen_anc`: Genetic ancestry group associated with the value `grpmax` above.
+- `faf99_joint`: A struct containing joint (exome + genome) FAF (99% CI).
+- `colocated_variants`: Array containing variants that are located at the same locus as this variant.
```
Yep, will add a few words clarifying that.

rileyhgrant commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,238 @@
+- `gnomad_constraint`: Struct containing gnomAD constraint information for gene. Struct is only present on the GRCh37 Hail Table.
+  - `gene`: Gene name.
+  - `transcript`: Transcript ID.
+  - `mane_select`: Whether transcript is the MANE Select transcript.
```
The v2 version (GRCh37 gene models) currently has gnomAD v2 constraint annotated.

A step was added to remove constraint from the v4 (GRCh38 gene models) due to the constraint being a bit more experimental.

Should we also remove constraint from the GRCh37/v2 gene model Hail Table?

ch-kr replied:
oh sorry I wasn't clear -- I meant that v2 constraint doesn't have `mane_select` annotated iirc (happy to be corrected, though). From my perspective, v2 gene constraint is stable and won't change, and it seems valuable to include in the gene models tables for our users (so fine with me to keep gene constraint in the GRCh37 table but remove from GRCh38 table)

rileyhgrant replied:
Ohhh I see oops I misunderstood. 

Good catch, will correct the fields here.

rileyhgrant commented:
Hiya @ch-kr, this should be ready for your re-review, at your convenience. 

Apart from addressing the comments there's a few added things (downloads page with placeholder links, the [changelog post ](https://github.com/broadinstitute/gnomad-blog/pull/166), and documenting of the VRS annotations and the joint frequency annotation).

Thank you!!

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,245 @@
+      - `n_larger`: Number of individuals with a higher age than the highest bin.
+    - `hom`: Struct containing age distribution information for individuals homozygous for this variant. Structured to allow easy histogram creation. Contains same fields as `het` above.
+  - `filters`: Set containing variant QC filters. See `filters` description on the v4 Hail Tables [help page](v4-hts#filters).
+  - `quality_metrics`: Struct containing variant quality metric histograms information.
+    - `allele_balance`: Struct containing variant allele balance histograms information.
+      - `alt_adj`: Struct containing variant allele balance information calculated across high-quality genotypes. Contains same fields as other histogram structs. This data is displayed in the "Allele balance for heterozygotes" histogram in the browser's variant page.
+      - `alt_raw`: Struct containing variant allele balance information calculated across unadjusted genotypes. Contains same fields as other histogram structs.
+    - `genotype_depth`: Struct containing information used to display genotype depth (DP) histograms.
+      - `all_adj`: Struct containing DP information calculated using high-quality genotypes. Contains same fields as other histogram structs.
+      - `all_raw`: Struct containing DP information calculated across unadjusted genotypes. Contains same fields as other histogram structs.
+      - `alt_adj`: Struct containing DP information calculated using high-quality genotypes (variant carriers only). Contains same fields as other histogram structs.
+      - `alt_raw`: Struct containing DP information calculated across unadjusted genotypes (variant carriers only). Contains same fields as other histogram structs.
+    - `genotype_quality`: Struct containing information used to display genotype quality (GQ) histograms.
+      - `all_adj`: Struct containing GQ information calculated using high-quality genotypes. Contains same fields as other histogram structs.
+      - `all_raw`: Struct containing GQ information calculated across unadjusted genotypes. Contains same fields as other histogram structs.
+      - `alt_adj`: Struct containing GQ information calculated using high-quality genotypes (variant carriers only). Contains same fields as other histogram structs.
+      - `alt_raw`: Struct containing GQ information calculated across unadjusted genotypes (variant carriers only). Contains same fields as other histogram structs.
```
on line 94 below, you have this description:
```
See `histograms` description on the v4 Hail Tables [help page](v4-hts#joint-histograms).
```
you could apply that here as well to save space

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,245 @@
+  - `strand`: Transcript strand.
+  - `start`: Transcript genomic start position (position only).
+  - `stop`: Transcript genomic stop position (position only).
+  - `xstart`: Transcript genomic start position (format: chromsomeposition).
+  - `xstop`: Transcript genomic stop position (format: chromsomeposition).
```
```suggestion
  - `xstart`: Transcript genomic start position (format: chromosomeposition).
  - `xstop`: Transcript genomic stop position (format: chromosomeposition).
```
sorry, I must have missed these previously!

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,245 @@
+  - `feature_type`: Exon type (e.g., CDS).
+  - `start`: Exon genomic start position (position only).
+  - `stop`: Exon genomic stop position (position only).
+  - `xstart`: Exon genomic start position (format: chromsomeposition).
+  - `xstop`: Exon genomic start position (format: chromsomeposition).
```
```suggestion
  - `xstart`: Exon genomic start position (format: chromosomeposition).
  - `xstop`: Exon genomic start position (format: chromosomeposition).
```

ch-kr commented on browser/help/topics/v4-browser-hts.md:
```diff
@@ -0,0 +1,245 @@
+- `symbol_upper_case`: All-caps gene symbol.
+- `search_terms`: Set containing search terms associated with gene.
+- `reference_genome`: Reference genome build associated with this gene.
+- `flags`: Set containing gene flags this gene.
```
```suggestion
- `flags`: Set containing gene flags for this gene.
```
(or something else; please correct if my suggestion is wrong)

ch-kr commented on browser/src/DownloadsPage/GnomadV2Downloads.tsx:
```diff
@@ -284,6 +286,29 @@ const GnomadV2Downloads = () => {
+          <Link to="/help/v4-browser-hts">help text</Link>
+        </p>
+
+        <h3>Browser gene models</h3>
```
naive question, does a browser variants for v2 exist?

rileyhgrant replied:
Ah good question, it does yes, every gnomAD release of variants also has a browser version. We could release the v2 browser variants table too if we think that would be useful to the wider community.

ch-kr replied:
ah, OK. I vote we stick to releasing only the v4 version for now and seeing if there is demand for the v2 table

ch-kr commented on data-pipeline/src/data_pipeline/data_types/gene.py:
```diff
@@ -226,6 +226,15 @@ def import_hgnc(path):
+    if keep_mane_version_global_annotation:
+        globals_dict = ds.index_globals()
+        ds = ds.select_globals(mane_select_version=globals_dict["annotations"]["mane_select_transcript"]["version"])
+    ds = ds.select_globals()
```
this `select_globals` statement needs to be in an `else` or the globals kept from line 233 will be removed, right?

ch-kr reviewed (approved):
thank you for adding the downloads! a few more minor things, otherwise LGTM. the one thing that needs updating is in `data-pipeline/src/data_pipeline/data_types/gene.py`

rileyhgrant commented on data-pipeline/src/data_pipeline/pipelines/gnomad_v4_variants.py:
```diff
@@ -102,6 +103,15 @@
+pipeline.add_task(
+    name="prepare_table_for_release",
+    task_function=prepare_table_for_release,
+    output_path=f"{output_sub_dir}/gnomad_v4_variants_a4_for_release.ht",
```
Ah, yeah I was being a bit silly and abbreviating unnecessarily (and confusingly) as well as wanting to keep the 'annotated_4' around somewhere in the filename, which it probably doesn't need, I'll rename it to something else.
